src/main/java/org/opentripplanner/street/search/state/StateData.java

```diff
@@ -51,10 +51,10 @@ public class StateData implements Cloneable {
 
   /** Private constructor, use static methods to get a set of initial states. */
   private StateData(StreetMode requestMode) {
-    if (requestMode.includesDriving()) {
-      currentMode = TraverseMode.CAR;
-    } else if (requestMode.includesWalking()) {
+    if (requestMode.includesWalking()) {
       currentMode = TraverseMode.WALK;
+    } else if (requestMode.includesDriving()) {
```

Comment:
Isn't this dependent on the direction?

Reply:
Pretty much spaghetti.

Reply:
Actually, can you add tests for car pickup and dropoff as well, this might break car dropoff, where we start in a car, and continue with walking?

Reply:
It might be easier to just have a switch-case here, explicitly listing all the street modes.

Reply:
That's a good idea. I've implemented this and added tests for CAR_PICKUP.

Could you have a quick look to check that the tests for CAR_PICKUP really assert the correct thing?